canvas_site_creator/views.py

```diff
@@ -27,6 +42,207 @@
 def lti_auth_error(request):
     raise PermissionDenied
 
+
+@login_required
+@lti_permission_required(canvas_api_accounts.ACCOUNT_PERMISSION_MANAGE_COURSES)
+@require_http_methods(['GET'])
+def index(request):
+    start_time = time.time()
+
+    sis_account_id = request.LTI['custom_canvas_account_sis_id']
+    ci_filters = {key: request.GET.get(key, '')
+                  for key in COURSE_INSTANCE_FILTERS}
+    (account_type, local_id) = canvas_api_accounts.parse_canvas_account_id(sis_account_id)
+    schools = []
+    terms = []
+    departments = []
+    course_groups = []
+    school = None
+
+    # Proceed only for valid account types
+    if account_type in canvas_api_accounts.ACCOUNT_TYPES:
+
+        # If user is in context Department or Course Group, display a
+        # message conveying restricted access
+        if (account_type == canvas_api_accounts.ACCOUNT_TYPE_DEPARTMENT or
+            account_type == canvas_api_accounts.ACCOUNT_TYPE_COURSE_GROUP):
+            return render(request, 'canvas_site_creator/restricted_access.html',
+                          status=403)
+
+        # We are in the context of a SIS type account, so limit options to that
+        # context
+        s = School.objects.get(pk=local_id)
+        school = {
+            'id': sis_account_id,
+            'name': s.title_short,
+        }
+
+        ci_filters[canvas_api_accounts.ACCOUNT_TYPE_SCHOOL] = school['id']
+        schools.append(school)
+
+    # if schools size is zero(includes improperly formatted account_type),
+    # display unauthorized message
+    if len(schools) == 0:
+        return redirect('not_authorized')
+    if school:
+        # Populate term, department, and course_group filter options if we already have a school
+        school_sis_account_id = school['id']
+
+        terms, current_term_id = get_term_data_for_school(school_sis_account_id)
+        if current_term_id:
+            ci_filters['term'] = current_term_id
+        if not departments and not course_groups:
+            departments = get_department_data_for_school(school_sis_account_id)
+            course_groups = get_course_group_data_for_school(school_sis_account_id)
+    context = {
+        'course_groups': course_groups,
+        'departments': departments,
+        'filters': ci_filters,
+        'in_school_account': (account_type ==
+                                  canvas_api_accounts.ACCOUNT_TYPE_SCHOOL),
+        'schools': schools,
+        'terms': terms,
+    }
+    logger.debug(" --------->Initial load of the Site creator view took : %s seconds" % (time.time() - start_time))
+    return render(request, 'canvas_site_creator/index.html', context)
+
+
+@login_required
+@lti_permission_required(canvas_api_accounts.ACCOUNT_PERMISSION_MANAGE_COURSES)
+@require_http_methods(['GET'])
+def audit(request):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 17. By comparison, 98% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 119-124 into a separate function.